src/networking.c

```diff
@@ -2760,6 +2760,12 @@ void processInlineBuffer(client *c) {
 static void setProtocolError(const char *errstr, client *c) {
     if (server.verbosity <= LL_VERBOSE || c->flag.primary) {
         sds client = catClientInfoString(sdsempty(), c, server.hide_user_data_from_log);
+        int loglevel = (c->flag.primary) ? LL_WARNING : LL_VERBOSE;
+
+        if (server.hide_user_data_from_log) {
+            printf("Query buffer during protocol error: *redacted*");
```

Comment:
Printf seems wrong. Did you mean snprintf?

```suggestion
            snprintf(buf, sizeof(buf), "Query buffer during protocol error: *redacted*");
```

Reply:
Since using goto after this line, there has no buf declared/initialed, I don't think we should use `snprintf`.

Reply:
`printf` isn't useful. Do you want to add a statement to logs? Try `serverLog` then.

Reply:
Got it. Will use the serverLog instead then

Reply:
So now we're logging twice? First here, then goto and log again?

Let's log only once.

Reply:
And if we don't log it here, there's no reason to move the line `int loglevel = (c->flag.primary) ? LL_WARNING : LL_VERBOSE;`. It can stay where it was.

Reply:
Okay, so based on the comments, the log afterwards would look like following:

When `server.hide_user_data_from_log` is off, 
```
<timestamp># WARNING: Receiving inline protocol from primary, primary stream corruption? Closing the primary connection and discarding the cached primary.
<timestamp># Protocol error (<protocol error message>) from client: <client> <query buffer info>. Query buffer during protocol error: <query buffer info>
```

When `server.hide_user_data_from_log` is on, 
```
<timestamp># WARNING: Receiving inline protocol from primary, primary stream corruption? Closing the primary connection and discarding the cached primary.
<timestamp># Protocol error (<protocol error message>) from client: <client>*redacted*. 
```

Reply:
Removed the snprint in latest version as suggested